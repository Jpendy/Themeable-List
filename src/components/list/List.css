.light {
    list-style: none;
    background: lightskyblue;
}

.dark {
    list-style: none;
    background: darkblue;
    color: white;
}
