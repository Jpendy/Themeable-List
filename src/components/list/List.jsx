import React from 'react';
import PropTypes from 'prop-types';

const List = ({ characters }) => {

  const listElement = characters.map((item, i) => {
    return (
      <li key={i} >
        <img src={item.image} />
        <p>{item.name}</p>
        <p>Species: {item.species}</p>
        <p>Status: {item.status}</p>
      </li>
    );
  });

  return (
    <ul>
      {listElement}
    </ul>
  );
};

List.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired
};

export default List;
