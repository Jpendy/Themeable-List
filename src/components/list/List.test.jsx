import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List characters={[{ image: 'image', name: 'Rick', species: 'human', status: 'alive' }, { image: 'image', name: 'Morty', species: 'human', status: 'alive' }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
