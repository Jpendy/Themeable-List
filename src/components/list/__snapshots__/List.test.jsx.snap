// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<ul>
  <li
    key="0"
  >
    <img
      src="image"
    />
    <p>
      Rick
    </p>
    <p>
      Species: 
      human
    </p>
    <p>
      Status: 
      alive
    </p>
  </li>
  <li
    key="1"
  >
    <img
      src="image"
    />
    <p>
      Morty
    </p>
    <p>
      Species: 
      human
    </p>
    <p>
      Status: 
      alive
    </p>
  </li>
</ul>
`;
